### DISCUSSION TITLE
Grafana dashboard for milvus with multi-tenancy strategies

### DISCUSSION THREAD

**Discussion by kish5430** [milvus-io/milvus#31907] [Ideas & Feature requests]
HI Team,

We have implemented multi-tenancy strategies in milvus, but we can currently view overall cluster resource utilization, we're unable to see the database level utilization.  Could you please add a feature that allows us to see database-level stats or a Grafana dashboard? This would enable us to analyze resource usage at the database level.

Thanks

**Comment by xiaofan-luan** [comment]
feel free to open a github issue for that. This could be a potential feature to improve

**Reply by kish5430** [reply]
I mean there is no issue , we need this feature to see the resource utilisation at database level.

**Reply by xiaofan-luan** [reply]
Sure I understand. We just create track features in github issue as well~

**Reply by xiaofan-luan** [reply]
BTW, I'd like to learn about your use case and feature requirment if you'd like to share.
My email is james.luan@zilliz.com and we definitely want to hep

**Reply by kish5430** [reply]
We want to implement Multi-tenancy strategies within Milvus. Additionally, it's important for us to track resource utilization individually

**Comment by xiaofan-luan** [comment]
Sure, we should be working on collection level metrics and db level metrics. 

@LoveEachDay

**Reply by kish5430** [reply]
Also can you plz add collection and database name for metric "milvus_proxy_req_count".

**Reply by xiaofan-luan** [reply]
I'm not sure how we can display collection level metrics on graphana. Anyone could help on this would be great

**Reply by xiaofan-luan** [reply]
We need collections, 
collection insert, delete, query, search, upsert QPS/latency  and entity numbers

**Comment by LoveEachDay** [comment]
@kish5430 Milvus 2.4 has added `database` level metrics. And we'll refine the grafana dashboard in the next coming week.

**Reply by kish5430** [reply]
@LoveEachDay Has Milvus added 'database' level metrics universally or just for selected metrics?

**Reply by kish5430** [reply]
@LoveEachDay Did you refine the grafana dashboard?

**Reply by kish5430** [reply]
HI @LoveEachDay Could you please provide ETA for releasing the refined Grafana dashboard?

**Reply by LoveEachDay** [reply]
@kish5430 Only for selected metrics with database name label. We are adding more metrics with database name label. We will release the new dashboard alongside milvus 2.4.1 release.

**Reply by kish5430** [reply]
@LoveEachDay It appears that the Milvus 2.4.1 version now includes a database label in the metrics for delete and bulk insert operations. Could you please also include collection creation,load, search and query operations?

 * Added db label to the metrics for the delete and bulk insert operations(https://github.com/milvus-io/milvus/pull/32611)

**Reply by kish5430** [reply]
HI @LoveEachDay Could you provide an update on adding metrics for collection creation, loading, searching, and querying operations?

**Reply by yhmo** [reply]
The work is still in progress, stay tuned.

**Reply by kish5430** [reply]
> The work is still in progress, stay tuned.

@yhmo Thank you for reply....Are you going to add dbname, collection name at cpu and memory usage metrics? if not, can you plz add

Thanks

**Reply by kish5430** [reply]
@yhmo @LoveEachDay Is there any ETA to release the new grafana dashboard?

**Reply by LoveEachDay** [reply]
@kish5430 We've refine the milvus dashboard, check it out from https://github.com/milvus-io/milvus/blob/master/deployments/monitor/grafana/milvus-dashboard.json

**Reply by kish5430** [reply]
> @kish5430 We've refine the milvus dashboard, check it out from https://github.com/milvus-io/milvus/blob/master/deployments/monitor/grafana/milvus-dashboard.json

@LoveEachDay I reviewed the new dashboard and noticed a few issues:

1. Collections are not added at the query level.
2. It only shows collections from the default database, which could be problematic for a multi-tenancy cluster.

To address this, please add a database filter before the collections filter and incorporate collections into as many charts as possible.

Thank you.
